File: WIP - Progetto/app/src/main/java/it/wip/database/dao/StoryDao.kt
package it.wip.database.dao

import androidx.room.*
import it.wip.database.model.Story
import it.wip.database.model.User

@Dao
interface StoryDao {

    @Query("SELECT * FROM story")
    suspend fun getAll(): Array<Story>

    @Query("SELECT * FROM story WHERE user = :userId")
    fun getAllByUserWithoutCoroutines(userId: Int): Array<Story>

    @Query("SELECT * FROM story WHERE id = :storyId")
    fun getAllByIdStoryWithoutCoroutines(storyId: Int): Array<Story>

    @Insert
    suspend fun insert(vararg stories: Story)

    @Insert
    fun insertWithoutCoroutines(vararg stories: Story)

    @Update
    suspend fun update(user: Story)

    @Delete
    suspend fun delete(user: Story)

}

File: wip_flutter/lib/database/dao/story_dao.dart
import 'package:sqflite/sqflite.dart';

import '../model/story.dart';

class StoryDao {

  static Future<List<Story>> getAll(Database db) async {
    final List<Map<String, dynamic>> maps = await db.query('story');
    return List.generate(maps.length, (i) {
      return Story(
          id: maps[i]['id'],
          storyName: maps[i]['story_name'],
          createdOn: maps[i]['created_on'],
          user: maps[i]['user']
      );
    });
  }

  static Future<void> insert(Database db, Story story) async {
    await db.insert('story', story.toMap());
  }

  static Future<void> update(Database db, Story story) async {
    await db.update('story', story.toMap(), where: 'id = ?', whereArgs: [story.id]);
  }

  static Future<void> delete(Database db, int id) async {
    await db.delete('story', where: 'id = ?', whereArgs: [id]);
  }

}